elasticdl/python/worker/task_data_service.py

```diff
@@ -83,6 +83,8 @@ def report_record_done(self, count, err_msg=""):
         if err_msg:
             self._failed_record_count += count
 
+        if not self._pending_tasks:
+            return False
         task = self._pending_tasks[0]
```

Comment:
Try to add a lock for self._pending_tasks[0] access.

Reply:
@skydoorkai I add a lock here, but the error still exists.